### PR TITLE
skip deltaBeat if dn is zero

### PR DESCRIPTION
**Pull request by kmlebedev**
# What problem are we solving?

https://github.com/seaweedfs/seaweedfs/issues/3629

# How are we solving the problem?

https://github.com/seaweedfs/seaweedfs/issues/3629

# How is the PR tested?

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by kmlebedev** [comment]
@chrislusf This is a critical bug and seems to have appeared in version 3.25 or 3.26 not sure why it used to work fine.

**Comment by kmlebedev** [comment]
@chrislusf I still don’t like the logic where the master sends a message to all clients with the removal of all volumes when leaving the cluster.

It seems that this is superfluous, and so, when changing the leader, we reset the volume map on the client.
What do you think?

**Comment by kmlebedev** [comment]
@chrislusf Ping

File: weed/server/master_grpc_server.go [hunk 1]
```diff
@@ -126,6 +127,10 @@ func (ms *MasterServer) SendHeartbeat(stream master_pb.Seaweed_SendHeartbeatServ
 
 		ms.Topo.Sequence.SetMax(heartbeat.MaxFileKey)
 		if dn == nil {
+			// skip deltaBeat
+			if heartbeat.Ip == "" {
```
Comment (chrislusf) [inline weed/server/master_grpc_server.go:131]: Add a warning message here. It should never happen after this PR.

**Comment by chrislusf** [comment]
> the master sends a message to all clients with the removal of all volumes when leaving the cluster.

which part of code are you referring to?

**Comment by kmlebedev** [comment]
> > the master sends a message to all clients with the removal of all volumes when leaving the cluster.
> 
> which part of code are you referring to?

https://github.com/seaweedfs/seaweedfs/blob/31d2f77ceb8aa0acd4ebf7a85ca24c9375de0a17/weed/server/master_grpc_server.go#L86-L91

After adding the cache(https://github.com/seaweedfs/seaweedfs/pull/3350), it no longer makes sense

**Pull request merged**